src/Features/Core/Portable/EditAndContinue/Remote/RemoteDebuggingSessionProxy.cs

```diff
@@ -10,13 +10,19 @@
 using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Remote;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.EditAndContinue;
 
-internal sealed class RemoteDebuggingSessionProxy(SolutionServices services, IDisposable? connection, DebuggingSessionId sessionId) : IActiveStatementSpanFactory, IDisposable
+internal sealed class RemoteDebuggingSessionProxy(
+    SolutionServices services,
+    IReferenceCountedDisposable<IDisposable> connection,
```

Comment:
Why passing in IReferenceCountedDisposable instead of the actual connection value like before?

Reply:
ReferenceCountedDisposal can't be variant.  And it wraps a specific type (not an IDisposable).  The IRefCountedDisposable exists for this pattern where we want to pass a base interface type.

Reply:
> Once the constructor is reached it owns the object

This is sometimes the case, but sometimes not. It also doesn't work at all for asynchronous code that uses resources. RCD provides stronger/clearer semantics in the face of ownership transfers and asynchronous/concurrent operations.